### PR TITLE
[JS] Topic: Iterables, Sets and Maps 

### PR DESCRIPTION
**Pull request by SakshiMishra1**
<hr>

## Description 📜

Clear explanation of iterables sets and maps with correct use of comments and code.
Fixes # <1833> 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [x] JavaScript
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi @s4-2 Please review. This PR is a recreation of #1881 .

File: JavaScript/Topics/Data_Structures/README.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+This is wonderful because the WeakSets objects set to null are garbage-collected while the program is still running, hence, reducing memory consumption and preventing memory leakage, especially when dealing with huge amounts of data that are generated asynchronously.
+
+Within this feature lies a chance for you to write light-weight solutions to programming problems without having to bother with the details of memory management.
+
+# Conclusion
+
+we’ve taken a look at Sets and Maps and how they handle unique items and key-value pairs respectively. These useful data structures provide easier and more efficient ways to structure and access data under certain use cases.
+
+*Special modifications such as WeakSets and WeakMaps provide more options for the developer and are handy for memory management.*
```
Comment (paramsiddharth) [inline JavaScript/Topics/Data_Structures/README.md:299]:
```suggestion
we’ve taken a look at Sets and Maps and how they handle unique items and key-value pairs respectively. These useful data structures provide easier and more efficient ways to structure and access data under certain use cases.

*Special modifications such as WeakSets and WeakMaps provide more options for the developer and are handy for memory management.*

__Contributor :__ [Sakshi Mishra](https://github.com/SakshiMishra1) :heart: 
```

Mention yourself as a contributor at the end of the file.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

**Comment by SakshiMishra1** [comment]
Sir I have modified the README.md file as per the changes you said. Please review it @paramsiddharth

**Comment by dakshsethi** [comment]
Looks good, but let's break it down into 2 subfolders for Maps and Sets
Where they each contain their own README.md with proper explanation
The main README.md can be kept short.

**Comment by SakshiMishra1** [comment]
> Looks good, but let's break it down into 2 subfolders for Maps and Sets
> Where they each contain their own README.md with proper explanation
> The main README.md can be kept short.

Ok sir, working on the changes.

**Comment by SakshiMishra1** [comment]
@dakshsethi Sir please review. Made the changes.
@paramsiddharth

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi @s4-2 Please review.

File: JavaScript/Topics/Data_Structures/README.md [hunk 2]
```diff
@@ -1,295 +1 @@
 # Iterables, Sets and Maps in JavaScript objects
-
-## Maps:-
```
Comment (dakshsethi) [inline JavaScript/Topics/Data_Structures/README.md]: Atleast explain something about Maps, Sets and Iterables in this README.md file
Then you can give the link to their respective files.

**Review by dakshsethi — CHANGES_REQUESTED** [review]

File: JavaScript/Topics/Data_Structures/README.md (hunk 2 above)
Comment (SakshiMishra1) [reply JavaScript/Topics/Data_Structures/README.md]: Done sir.

**Comment by SakshiMishra1** [comment]
@paramsiddharth @dakshsethi please review sir. All changes made.

File: JavaScript/Topics/Data_Structures/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+## What is a MAP?
+
+ A JavaScript Map is an associative collection of distinct keys and values. Both keys and values can be any primitive or object. This is a significant data structure with many valuable uses.
+
+ ## What is set?
+
+ Set is another new collection introduced by ES6. The Array class of JavaScript can work with ordered collection of data, but not so well with unordered collections or when values held in the collection are unique. That’s why JavaScript introduced Set.
+
+  _a Map is a collection of ordered key/value pairs, and that a Set is a collection of unique values. Both of these data structures add additional capabilities to JavaScript and simplify common tasks such as finding the length of a key/value pair collection and removing duplicate items from a data set, respectively. On the other hand, Objects and Arrays have been traditionally used for data storage and manipulation in JavaScript, and have direct compatibility with JSON, which continues to make them the most essential data structures, especially for working with REST APIs. Maps and Sets are primarily useful as supporting data structures for Objects and Arrays._
```
Comment (dakshsethi) [inline JavaScript/Topics/Data_Structures/README.md:11]: Map definition should be under `What is a MAP?` section

**Review by dakshsethi — CHANGES_REQUESTED** [review]

File: JavaScript/Topics/Data_Structures/README.md (hunk 3 above)
Comment (SakshiMishra1) [reply JavaScript/Topics/Data_Structures/README.md:11]: Sir, it's a kind of conclusion of what the folder contains (line number 11).
Should I still change it?
@dakshsethi

Comment (dakshsethi) [reply JavaScript/Topics/Data_Structures/README.md:11]: Ok then just give one more space and at least write the heading as Conclusion or something
Also mention `**Contributor:** [Sakshi Mishra](https://github.com/SakshiMishra1)` at the end of every page

Comment (SakshiMishra1) [reply JavaScript/Topics/Data_Structures/README.md:11]: Ok sir, making the changes.

Comment (SakshiMishra1) [reply JavaScript/Topics/Data_Structures/README.md:13]: Done with the changes sir. @dakshsethi

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥🔥

**Comment by paramsiddharth** [comment]
@Creatoon @s4-2 Please review the PR.

**Review by paramsiddharth — APPROVED** [review]

**Comment by paramsiddharth** [comment]
image

**Pull request closed without merging**